src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -789,6 +795,7 @@ class NBDListIterator {
 
       *cfg = Config();
       cfg->devpath = "/dev/nbd" + stringify(m_index++);
+      cfg->cookie = get_cookie(cfg->devpath);
```

Comment:
Why are we querying the kernel for the cookie here?  It looks like the config (along with the cookie) may already be cached in `m_mapped_info_cache`.  Also, if we do it in `get_mapped_info()`, `find_attached()` wouldn't need a cookie parameter.  I would move this call to line 912 and possibly validate that the cookie from `parse_args()` (if any) matches the one from the kernel.

Reply:
looks reasonable to move it under get_mapped_info()